catalog/core/catalog-core-contentresourcereader/src/main/java/org/codice/ddf/catalog/content/resource/reader/DerivedContentActionProvider.java

```
@@ -0,0 +1,114 @@
+        if (!canHandle(input)) {
+            return Collections.emptyList();
+        }
+        // Expect only 1
+        List<Action> resourceActions = resourceActionProvider.getActions(input);
+        if (resourceActions.isEmpty()) {
+            return Collections.emptyList();
+        }
+        List<Action> actions = new ArrayList<>();
+        ((Metacard) input).getAttribute(Metacard.DERIVED_RESOURCE_URI)
+                .getValues()
+                .stream()
+                .forEach(value -> {
+                    try {
+                        URI uri = new URI(value.toString());
+                        URIBuilder builder = new URIBuilder(resourceActions.get(0)
+                                .getUrl()
+                                .toURI());
+                        if (StringUtils.equals(uri.getScheme(), ContentItem.CONTENT_SCHEME)) {
+                            String qualifier = uri.getFragment();
+
+                            builder.addParameters(Arrays.asList(new BasicNameValuePair("options",
+                                    qualifier)));
+                            ActionImpl newAction = new ActionImpl(ID,
+                                    "View " + qualifier,
+                                    DESCRIPTION_PREFIX + qualifier,
+                                    builder.build()
+                                            .toURL());
+                            actions.add(newAction);
+                        } else {
+                            ActionImpl newAction = new ActionImpl(ID,
+                                    "View " + uri.toString(),
+                                    DESCRIPTION_PREFIX + uri.toString(),
+                                    uri.toURL());
+                            actions.add(newAction);
+                        }
+                    } catch (URISyntaxException | MalformedURLException e) {
+                        LOGGER.debug("Unable to create action URL.", e);
```

Should this be debug level or warn? At debug level, most of these exceptions will get swallowed up in a prod environment.